.github/workflows/publish.yml

```
@@ -0,0 +1,95 @@
+
+        # botify will update the version on `main`, so this check is important to prevent an infinite loop
+        if: ${{ github.actor != 'botify' }}
+
+        steps:
+            # Running this action ensures that only one instance of this job will run at a time.
+            # This is important to prevent race conditions when multiple pull requests are merged in quick succession
+            - uses: softprops/turnstyle@8db075d65b19bf94e6e8687b504db69938dc3c65
+              with:
+                  poll-interval-seconds: 10
+              env:
+                  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+            - name: Decrypt & Import Botify GPG key
+              run: |
+                cd .github
+                gpg --quiet --batch --yes --decrypt --passphrase="$LARGE_SECRET_PASSPHRASE" --output Botify-private-key.asc Botify-private-key.asc.gpg
+                gpg --import --batch Botify-private-key.asc
+              env:
+                LARGE_SECRET_PASSPHRASE: ${{ secrets.LARGE_SECRET_PASSPHRASE }}
+
+            - name: Set up git for Botify
+              run: |
+                git config --global user.signingkey F34C92C6A7FB2652
+                git config --global commit.gpgsign true
+                git config --global user.name botify
+                git config --global user.email infra+botify@expensify.com
+
+            - uses: actions/checkout@v3
+
+            - uses: actions/setup-node@v3
+              with:
+                node-version: '16.x'
+                registry-url: 'https://registry.npmjs.org'
+
+            - name: Generate branch name
+              run: echo "BRANCH_NAME=botify-bump-version-$(uuidgen)" >> $GITHUB_ENV
+
+            - name: Create branch for version-bump pull request
+              run: git checkout -b ${{ env.BRANCH_NAME }}
+
+            - name: Install npm packages
+              run: npm ci
+
+            - name: Update npm version
+              run: npm version patch
+
+            - name: Set new version in GitHub ENV
+              run: echo "NEW_VERSION=$(npm info react-native-onyx version)" >> $GITHUB_ENV
+
+            - name: Build package
+              run: npm run build
+
+            - run: npm publish
+              env:
+                NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+            - name: Push branch and publish tags
+              run: git push --tags
+
+            - name: Create pull request
+              id: createPullRequest
+              # Version: 2.6.2
+              uses: repo-sync/pull-request@65785d95a5a466e46a9d0708933a3bd51bbf9dde
+              with:
+                source_branch: ${{ env.BRANCH_NAME }}
+                destination_branch: main
+                github_token: ${{ secrets.BOTIFY_TOKEN }}
+                pr_title: Update version to ${{ env.NEW_VERSION }}
+                pr_body: 'Update version to ${{ env.NEW_VERSION }}'
+                pr_label: automerge
+
+            - name: Auto-approve pull request
```

Again, let's just use the gh cli (note that the approver and PR creator must be different tokens)
```suggestion
            - name: Auto-approve pull request
              run: gh pr review --approve
              env:
                GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```